PluginsAndFeatures/azure-toolkit-for-intellij/resources/hdinsight/templates/scala/sample/AgeMean_Div0.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package sample
+
+import org.apache.spark.{SparkConf, SparkContextWithFailureSave}
+import org.apache.spark.sql.{Row, SparkSession}
+
+case class GenderMeanAge(gender: String, meanAge: Int)
+case class People(name: String, age: Int, gender: String)
+object AgeMean_Div0 {
+
+  def main(args: Array[String]) {
+    val sparkconf = new SparkConf().setAppName("SQL Query with issue")
+
+    val sc = new SparkContextWithFailureSave(sparkconf)
+    val spark = SparkSession
+      .builder()
+      .appName("Spark Age Mean")
+      .getOrCreate()
```

Comment:
If we use SparkContext, SparkSession is useless here?

Reply:
The created SparkContext is to force SparkSession to use the SparkContextWithFailureSave one. Since we use the DataFrames API later, the SparkSession needs to be initialized here.

```
// For implicit conversions like converting RDDs to DataFrames
import spark.implicits._
```